### PR TITLE
Sinclair ZX Spectrum Multiface128 and diagnostic ROMs

### PR DESCRIPTION
**Pull request by rfka01**

### CODE REVIEW

File: src/devices/bus/spectrum/mface.cpp [hunk 1]
```diff
@@ -172,7 +172,9 @@ ROM_START(mface128)
 	ROM_REGION(0x2000, "rom", 0)
 	ROM_DEFAULT_BIOS("v363c")
 	ROM_SYSTEM_BIOS(0, "v363c", "87.2 V36 3C")
-	ROMX_LOAD("mf128_36_3c.rom", 0x0000, 0x2000, CRC(78ec8cfd) SHA1(8df204ab490b87c389971ce0c7fb5f9cbd281f14), ROM_BIOS(0))
+	ROMX_LOAD("mf128_36_3c.rom", 0x0000, 0x2000, CRC(78ec8cfd) SHA1(8df204ab490b87c389971ce0c7fb5f9cbd281f14), ROM_BIOS(0))
```
Comment (cuavas) [inline src/devices/bus/spectrum/mface.cpp:175]: Why does this line show as changed?  I’m always wary of invisible changes to lines.

Comment (rfka01) [reply src/devices/bus/spectrum/mface.cpp:175]: I was considering re-sorting the ROMs (putting the new one over the old one) but then decided against it.

Comment (cuavas) [reply src/devices/bus/spectrum/mface.cpp:175]: If there’s no net change to the line, it shouldn’t show as changed in the diff.  What’s actually changed in this line to cause it to be highlighted in the diff?

Comment (p1pkin) [reply src/devices/bus/spectrum/mface.cpp:175]: perhaps was added trailing space ?

on the side note - why not add 2 more remaining MF128 dumps listed there ? https://x128.speccy.cz/multiface/multiface.htm

Comment (rfka01) [reply src/devices/bus/spectrum/mface.cpp]: @p1pkin a trailing space it was, thanks! You led me down quite a rabbit hole with that link ... the actual ROMs are found elsewhere.
@cuavas building SRCCLEAN right now ,sorry ... :)

File: src/devices/bus/spectrum/mface.cpp [hunk 2]
```diff
@@ -173,6 +171,12 @@ ROM_START(mface128)
 	ROM_DEFAULT_BIOS("v363c")
 	ROM_SYSTEM_BIOS(0, "v363c", "87.2 V36 3C")
 	ROMX_LOAD("mf128_36_3c.rom", 0x0000, 0x2000, CRC(78ec8cfd) SHA1(8df204ab490b87c389971ce0c7fb5f9cbd281f14), ROM_BIOS(0))
+	ROM_SYSTEM_BIOS(1, "vxx0f", "87.2 Vxx 1D") // unknown PCB version
+	ROMX_LOAD("mf128_xx_1d.rom", 0x0000, 0x2000, CRC(f473991e) SHA1(f03f4ecbcf4a654f4775d16bda0d4cc47f884379), ROM_BIOS(1))
+	ROM_SYSTEM_BIOS(2, "v350d", "87.12 V35 0F")
+	ROMX_LOAD("mf128_35_0f.rom", 0x0000, 0x2000, CRC(cfefd560) SHA1(6cd6fd2c0fbb40a989a568db9d08ba8eed49cbbd), ROM_BIOS(1))
+	ROM_SYSTEM_BIOS(3, "v340d", "87.1 V34 0D")
+	ROMX_LOAD("mf128_34_0d.rom", 0x0000, 0x2000, CRC(8d8cfd39) SHA1(2104962bb6097e58fcab63969bbaca424a872bb5), ROM_BIOS(3))
```
Comment (cuavas) [inline src/devices/bus/spectrum/mface.cpp:179]: Please put these in ascending order to be consistent with the other one

File: src/mame/drivers/spectrum.cpp [hunk 3]
```diff
@@ -893,6 +893,24 @@ ROM_START(spectrum)
...
+	ROM_SYSTEM_BIOS(23, "diagv35", "DiagROM v1.35")
+	ROMX_LOAD("diagrom.v35",0x0000,0x4000, CRC(c869b642) SHA1(61eb786249a207ea57cdea6bb01fcd0b95c218f2), ROM_BIOS(23))
+	ROM_SYSTEM_BIOS(24, "diagv34", "DiagROM v1.34")
+	ROMX_LOAD("diagrom.v34",0x0000,0x4000, CRC(50478f29) SHA1(1e55af6e7bb24090ce052e7536e91f92492fe168), ROM_BIOS(24))
+	ROM_SYSTEM_BIOS(25, "diagv33", "DiagROM v1.33")
+	ROMX_LOAD("diagrom.v33",0x0000,0x4000, CRC(0fb080e1) SHA1(222f535ae277d0b8ce6c27b75319da9c45f8cadc), ROM_BIOS(25))
+	ROM_SYSTEM_BIOS(26, "diagv31", "DiagROM v1.31")
+	ROMX_LOAD("diagrom.v31",0x0000,0x4000, CRC(cddbb11e) SHA1(3ed203034fe1562ee7216fad06309d3f7033a967), ROM_BIOS(26))
+	ROM_SYSTEM_BIOS(27, "diagv28", "DiagROM v1.28")
+	ROMX_LOAD("diagrom.v28",0x0000,0x4000, CRC(aadf967e) SHA1(6fbdb11e475499655df36343a22f15d67f578165), ROM_BIOS(27))
```
Comment (cuavas) [inline src/mame/drivers/spectrum.cpp:911]: Please put these diagnostic ROMs in ascending order by version.  Other ROMs are in ascending version order here (BusySoft Upgrade and Turbo).

**Review by cuavas — CHANGES_REQUESTED** [review]

**Comment by Tafoid** [comment]
@rfka01 
Multiface 3 [folder: spectrum_mface3 - size: 24kb]
missing rom: mf3_hm_9a.rom [size: 8192] [CRC32: 2ce53095] [SHA1: 5fa286f1552f26575a14ab32125d59c26ce95978]

The issue here is the size is incorrect, showing 16384 (0x4000) which is a size unlike all the others.  The hashes provided match the image that was shared.

**Pull request merged**